packages/connect/src/base/stun.ts

```diff
@@ -300,44 +293,51 @@ function decodeIncomingSTUNResponses(addrs: Request[], socket: Socket, ms: numbe
   })
 }
 
-function getUsableResults(results: Request[], runningLocally = false): Request[] {
-  let filtered: Request[] = []
+/**
+ * Remove unusable responses from results
+ * @param responses results to filter
+ * @param runningLocally whether to run in local-mode or not
+ * @returns filtered results
+ */
+export function getUsableResults(responses: Request[], runningLocally = false): RequestWithResponse[] {
+  let filtered: RequestWithResponse[] = []
 
-  for (const result of results) {
-    if (!result.response) {
+  for (const result of responses) {
+    if (!isRequestWithResponse(result)) {
       continue
     }
 
-    switch (result.response.family) {
+    switch (result.response.family as 'IPv6' | 'IPv4') {
       case 'IPv6':
         // STUN over IPv6 is not yet supported
         break
       case 'IPv4':
         const u8aAddr = ipToU8aAddress(result.response.address, 'IPv4')
-        // Disitinguish two use cases:
-        // Unit tests:
-        // - run several instances on one machine, hence STUN response is expected to be
-        //   'localhost:somePort'
-        // CI tests / large E2E tests:
-        // - run several instances on multiple machines running in the *same* local network
-        //   hence STUN response is expected to be a local address
-        // Disclaimer:
-        // the mixed use case, meaning some instances running on the same machine and some
-        // instances running on machines in the same network is not expected
-        if ((isPrivateAddress(u8aAddr, 'IPv4') || isLocalhost(u8aAddr, 'IPv4')) == runningLocally) {
-          filtered.push(result)
+
+        if (runningLocally) {
+          // Only take local or private addresses
+          if (isPrivateAddress(u8aAddr, 'IPv4') || isLocalhost(u8aAddr, 'IPv4')) {
+            filtered.push(result)
+          }
+        } else {
+          // Only take public addresses
+          if (!isPrivateAddress(u8aAddr, 'IPv4') && !isLocalhost(u8aAddr, 'IPv4')) {
+            filtered.push(result)
+          }
         }
         break
-      default:
-        error(`Invalid STUN response. Got family: ${result.response.family}`)
-        break
     }
   }
 
   return filtered
 }
 
-function intepreteResults(results: Request[]):
+/**
+ * Check if the results are ambiguous and return single response
+ * if not ambiguous.
+ * @param results results to check for ambiguity
+ */
+export function intepreteResults(results: RequestWithResponse[]):
   | {
       ambiguous: true
```

Comment:
Can we perhaps make the return value more concise like :
````
{
  ambiguous: boolean
  publicAddress?: Address
}
````

Reply:
It is pretty concise, because it requires `publicAddress` to be defined if, and only if, the result is not ambiguous